roles/iam-user/tasks/main.yml

```diff
@@ -0,0 +1,18 @@
+---
+- block:
+    - name: Include create IAM group scripts
+      include: create_groups.yml
+      when: not destroy_stack|default(False)
```

Comment:
this will create the group even if it not wanted. perhaps you should only run the group task if the user dict is present.